wdns/str_to_rdata_ubuf.c

```diff
@@ -934,22 +971,121 @@ _wdns_str_to_rdata_ubuf(ubuf *u, const char *str,
 		}
 
 		case rdf_svcparams: {
-			/* process ony key=val pair at a time */
-			char *buf, *tok, *endp = NULL;
+			char *start, *eol;
+			int prev_key = -1;
+
+			eol = (char *)str;
```

Comment:
Casting `const char *` to `char *` should be avoided, as it bypasses an important safety measure in C.

That said, I do not see `start` or `eol` (or `key`, `val`, or `end` below) being used to modify the underlying string, so these should all be declared `const char *`.